src/discussions-trigger.ts

```diff
@@ -0,0 +1,175 @@
+    } else if (categoryID === "request-a-new-types-package" && info.action === "created") {
+        return updateDiscordWithRequest(info.body.discussion);
+    }
+
+    return reply(context, 204, "Can't handle this specific request");
+};
+
+function extractNPMReference(discussion: Discussion) {
+    const title = discussion.title;
+    if (title.includes("[") && title.includes("]")) {
+        return title.split("[")[1]!.split("]")[0];
+    }
+    return undefined;
+}
+
+const couldNotFindMessage = `Hi, we could not find a reference to the types you are talking about in this discussion. 
+Please edit the title to include the name on npm inside square brackets. 
+
+E.g. 
+ - \`"[@typescript/vfs] Does not x, y"\`
+ - \`"Missing x inside [node]"\`
+ - \`"[express] Broken support for template types"\`
+`;
+
+const gotAReferenceMessage = (module: string, owners: string[]) => `Thanks for the discussion about "${module}", some useful links for everyone: [npm](https://www.npmjs.com/package/${module}) / etc
+
+Pinging the DT module owners: ${owners.join(", ")}.
+`;
+
+
+async function pingAuthorsAndSetUpDiscussion(discussion: Discussion) {
+    const aboutNPMRef = extractNPMReference(discussion);
+    if (!aboutNPMRef) {
+        // https://gist.github.com/smashwilson/311e1487ddb40a455fc54d294cc63ad4#adddiscussioncomment
+        await updateOrCreateMainComment(discussion, couldNotFindMessage);
+        return;
+    } else {
+        const message = gotAReferenceMessage(aboutNPMRef, []);
+        await updateOrCreateMainComment(discussion, message);
+        await addLabel(discussion, aboutNPMRef, `Discussions related to ${aboutNPMRef}`);
+        return;
+    }
+}
+
+async function updateDiscordWithRequest(discussion: Discussion) {
+    const discordWebhookAddress = process.env.DT_MODULE_REQ_DISCORD_WEBHOOK;
+    if (!discordWebhookAddress) throw new Error("DT_MODULE_REQ_DISCORD_WEBHOOK not set in ENV");
+
+    // https://birdie0.github.io/discord-webhooks-guide/discord_webhook.html
+    const webhook = {  content: `New DT Module requested:`, embeds: [ { title: discussion.title, url: discussion.html_url } ] };
+    await fetch(discordWebhookAddress, { method: "POST", body: JSON.stringify(webhook), headers: { "content-type": "application/json" } });
+}
+
+
+async function updateOrCreateMainComment(discussion: Discussion, message: string) {
+    const discussionComments = await getCommentsForDiscussionNumber(discussion.number);
+    const previousComment = discussionComments.find(c => c.author.login === "typescript-bot");
+    if (previousComment) {
+        await client.mutate(createMutation<any>("updateDiscussionComment" as any, { body: message, commentId: previousComment.id }));
+    } else {
+        await client.mutate(createMutation<any>("addDiscussionComment" as any, { body: message, discussionId: discussion.node_id }));
+    }
+}
+
+async function addLabel(discussion: Discussion, labelName: string, description?: string) {
```

Comment:
Wouldn't it upset GH if we end up with up to thousands of labels?

But more importantly, it looks like they're project-global, so having these massive number of labels is going to make dealing with labels pretty difficult.  The least that this should do is use some prefix for these things, otherwise I can post a message with a `[Critical]` and confuse people and code...

Reply:
Yeah, I've opted for "Pkg: " - I'd like something a bit further in the alphabet if you can thing of something better though? (emoji/non-ascii aren't counted in the sorting for labels)

Reply:
I didn't see the `Pkg:` change, so leaving this open.  Also: I'm still worried about the first point: we have thousands of packages, and therefore we'd end up with thousands of labels, and I'm not sure how things would look like if/when we have that.